tests/drivers/hwinfo/api/src/main.c

```diff
@@ -86,11 +86,139 @@ static void test_device_id_enotsup(void)
 static void test_device_id_enotsup(void) {}
 #endif /* CONFIG_HWINFO_HAS_DRIVER not defined*/
 
+/*
+ * @addtogroup t_hwinfo_get_reset_cause_api
+ * @{
+ * @defgroup t_hwinfo_get_reset_cause test_hwinfo_get_reset_cause
+ * @brief TestPurpose: verify get reset cause works.
+ * @details
+ * - Test Steps
+ *   -# Set target buffer to a known value
+ *   -# Read the reset cause
+ *   -# Check if target buffer has been altered
+ * - Expected Results
+ *   -# Target buffer contents should be changed after the call.
+ * @}
+ */
+#ifdef CONFIG_HWINFO_HAS_DRIVER
+static void test_get_reset_cause(void)
+{
+	u32_t cause;
+	ssize_t ret;
+
+	/* Set `cause` to a known value prior to call. */
+	cause = 0xDEADBEEF;
+
+	ret = hwinfo_get_reset_cause(&cause);
+	/* If a platform does not support reset cause, pass the test. */
+	if (ret == -ENOTSUP) {
+		return;
+	}
+	// zassert_not_equal(ret, -ENOTSUP, "Not supported by hardware");
+	zassert_false((ret < 0),
+		      "Unexpected negative return value: %d", ret);
+
+	/* Verify that `cause` has been changed. */
+	zassert_not_equal(cause, 0xDEADBEEF, "Reset cause not written.");
+}
+#else
+static void test_get_reset_cause(void) {}
+#endif /* CONFIG_HWINFO_HAS_DRIVER */
+
+#ifndef CONFIG_HWINFO_HAS_DRIVER
+static void test_get_reset_cause_enotsup(void)
+{
+	ssize_t ret;
+	u32_t cause;
+
+	ret = hwinfo_get_reset_cause(&cause);
+	/* There is no hwinfo driver for this platform, hence the return value
+	 * should be -ENOTSUP
+	 */
+	zassert_equal(ret, -ENOTSUP,
+		      "hwinfo_get_reset_cause returned % instead of %d",
+		      ret, -ENOTSUP);
+}
+#else
+static void test_get_reset_cause_enotsup(void) {}
+#endif /* CONFIG_HWINFO_HAS_DRIVER not defined*/
+
+
+/*
+ * @addtogroup t_hwinfo_clear_reset_cause_api
+ * @{
+ * @defgroup t_hwinfo_clear_reset_cause test_hwinfo_clear_reset_cause
+ * @brief TestPurpose: verify clear reset cause works. This may
+ *        not work on some platforms, depending on how reset cause register
+ *        works on that SoC.
+ * @details
+ * - Test Steps
+ *   -# Read the reset cause and store the result
+ *   -# Call clear reset cause
+ *   -# Read the reset cause again
+ *   -# Check if the two readings match
+ * - Expected Results
+ *   -# Reset cause value should change after calling clear reset cause.
+ * @}
+ */
+#ifdef CONFIG_HWINFO_HAS_DRIVER
```

Comment:
Can you split `CONFIG_HWINFO_HAS_DRIVER` to `CONFIG_HWINFO_DEVICE_ID_HAS_DRIVER` and  `CONFIG_HWINFO_RESET_CAUSE_HAS_DRIVER`